db/main/migrate/20180329000000_create_github_installations.rb

```diff
@@ -0,0 +1,32 @@
+class CreateGithubInstallations < ActiveRecord::Migration
+  PSQL = ActiveRecord::Base.connection.select_value('SELECT version()')
+
+  def self.up
+    create_table :github_installations do |t|
+      t.belongs_to :owner, :polymorphic => true
+      t.string     :owner_type
+      t.integer    :owner_id
+      if /PostgreSQL 9.3/.match(PSQL)
+        t.json       :permissions
+      else
+        t.jsonb    :permissions
+      end
+      t.integer    :added_by
+      t.datetime   :deleted_on
+      t.datetime   :updated_on
+    end
+
+    def change
+      add_reference :github_installations, :added_by, foreign_key: { to_table: :users }
+      add_reference :github_installations, :deleted_by, foreign_key: { to_table: :users }
+      add_column :repositories, :active_on_org, :boolean
+      add_column :repositories, :activated_by_github_apps_on, :timestamp
+    end
+  end
```

Comment:
I haven't mixed `up/down` and `change` so I'm not sure if there's a guarantee of the order they fire in. I /might/ consider moving the reference and column additions to a 2nd migration, but it isn't a deal breaker to have them here.

Reply:
okay - will leave them here for now

Reply:
actually i think they need to be in a separate migration...